_episodes_rmd/05-data-visualization.Rmd

```diff
@@ -142,129 +133,163 @@ Building plots with **`ggplot2`** is typically an iterative process. We start by
+> >  geom_point() + labs(x="Base Pair Position", y="Mapping Quality (MQ)")
+> > ```
 > {: .solution}
 {: .challenge}
 
-## Boxplot
+## Faceting
 
-When we have one continuous and one categorical variable
-it is usually much more appropriate to make a boxplot
-than a scatterplot. In this case, our continuous variable
-is read depth (`DP`) and our categorical variable in
-`sample_id`. We can use boxplots to visualize the distribution of (filtered) sequencing depth `DP` within each sample:
+**`ggplot2`** has a special technique called *faceting* that allows the user to split one
+plot into multiple plots based on a factor included in the dataset. We will use it to split our
+mapping quality plot into three panels, one for each sample.
 
-```{r boxplot, purl=FALSE}
-ggplot(data = variants, aes(x = sample_id, y = DP))+
-geom_boxplot()
+```{r first-facet, purl=FALSE}
+ggplot(data = variants, aes(x = POS, y = MQ, color = sample_id)) + 
+ geom_point() + labs(x="Base Pair Position", y="Mapping Quality (MQ)") + 
+    facet_grid(~ sample_id)
 ```
 
-By adding points to boxplot, we can have a better idea of the number of
-measurements and of their distribution:
+This looks ok, but it would be easier to read if the plot facets were stacked vertically rather 
+than horizontally. The `facet_grid`
+geometry allows you to explicitly specify how you want your plots to be
+arranged via formula notation (`rows ~ columns`; a `.` can be used as
+a placeholder that indicates only one row or column).
 
-```{r boxplot-with-points, purl=FALSE}
-ggplot(data = variants, aes(x = sample_id, y = DP))+
-geom_boxplot() +
-geom_jitter(alpha = 0.3, color = "tomato")
+```{r second-facet, purl=FALSE}
+ggplot(data = variants, aes(x = POS, y = MQ, color = sample_id)) + 
+ geom_point() + labs(x="Base Pair Position", y="Mapping Quality (MQ)") + 
+    facet_grid(sample_id ~ .)
 ```
 
-> ## Exercise
-> 
-> In the plot above, the boxplot layer appears behind the 
-> jitter layer. Recreate the plot, but put the boxplot 
-> in front of the jitter plot so that it's easier to see
-> the boxplot.
->> ## Solution
->> ```{r}
->> ggplot(data = variants, aes(x = sample_id, y = DP)) +
->> geom_jitter(alpha = 0.3, color = "tomato") + 
->> geom_boxplot()
->> ```
-> {: .solution}
-{: .challenge}
+Usually plots with white background look more readable when printed.  We can set
+the background to white using the function `theme_bw()`. Additionally, you can remove
+the grid:
 
-> ## Exercise (intermediate)
->
-> Boxplots are useful summaries, but hide the *shape* of the distribution. For
-> example, if the distribution is bimodal, we would not see it in a
-> boxplot. An alternative to the boxplot is the violin plot, where the shape
-(of the density of points) is drawn.
+```{r facet-plot-white-bg, purl=FALSE}
+ggplot(data = variants, aes(x = POS, y = MQ, color = sample_id)) + 
+  geom_point() + labs(x="Base Pair Position", y="Mapping Quality (MQ)") + 
```

Comment:
Same as above, put labs on its own line and split `x` and `y` parameters across 2 lines.